api.planx.uk/pay/pay.ts

```diff
@@ -19,40 +20,86 @@ export async function makePaymentViaProxy(
   const isSupported =
     process.env[`GOV_UK_PAY_TOKEN_${req.params.localAuthority.toUpperCase()}`];
 
+  if (!isSupported) {
+    return next({
+      status: 400,
+      message: `GOV.UK Pay is not enabled for this local authority`,
+    });
+  }
+
   const flowId = req.query?.flowId as string | undefined;
   const sessionId = req.query?.sessionId as string | undefined;
   const teamSlug = req.params.localAuthority;
 
-  if (isSupported) {
+  // drop req.params.localAuthority from the path when redirecting
+  // so redirects to plain [GOV_UK_PAY_URL] with correct bearer token
+  usePayProxy(
+    {
+      pathRewrite: (path) => path.replace(/^\/pay.*$/, ""),
+      selfHandleResponse: true,
+      onProxyRes: responseInterceptor(
+        async (responseBuffer, _proxyRes, _req, _res) => {
+          const responseString = responseBuffer.toString("utf8");
+          const govUkResponse = JSON.parse(responseString);
+          await logPaymentStatus({
+            sessionId,
+            flowId,
+            teamSlug,
+            govUkResponse,
+          });
+          return responseBuffer;
+        }
+      ),
+    },
+    req
+  )(req, res, next);
+}
+
+export async function makeInviteToPayPaymentViaProxy(req: Request, res: Response, next: NextFunction) {
```

Comment:
I did spend a bit of time attempting to follow the pattern of creating a function which just accepted a callback instead of being invite to pay specific without much luck.

The complexity here (and one of the reasons that I suspect `fetchPaymentViaProxyWithCallback()` exists) is that `http-proxy-middleware` must be the _last_ middleware layer (as it calls `res.send()`). 

There's a good thread on this here alongside a few possibly more generic solutions - https://github.com/chimurai/http-proxy-middleware/issues/248